common/sphinx.c

```diff
@@ -198,43 +197,29 @@ static void xor_cipher_stream(void *dst, const struct secret *k, size_t dstlen)
 	crypto_stream_chacha20_xor(dst, dst, dstlen, nonce, k->data);
 }
 
-static bool compute_hmac(
-	void *dst,
-	const void *src,
-	size_t len,
-	const void *key,
-	size_t keylen)
+static void compute_hmac(const u8 *src, size_t slen,
```

Comment:
This wrapper doesn't do much, maybe we can remove it as well?

Reply:
Will do in another PR.

Reply:
Actually, it is kinda useful since it takes a secret.